eth/abc.py

```diff
@@ -74,6 +74,8 @@ class MiningHeaderAPI(ABC):
     gas_used: int
     timestamp: int
     extra_data: bytes
+    gas_target: int  # EIP-1559
+    base_fee_per_gas: int  # EIP-1559
```

Comment:
This feels... unfortunate, but it might be necessary.  Ideally, it would be nice if these fields didn't leak into all the previous VM APIs.... but doing so might introduce even more unfortunate complexity into our type definitions which might be worse...

Reply:
I agree and will take a deeper look -- for now I can say a similar problem applies to transactions. Base `transaction.validate` methods all type-check `gas_price` which is no longer a thing in London transactions so they need to be duplicated almost entirely. We could keep `gas_price` as an (internal, not exposed) alias for either `max_fee_per_gas` or `max_priority_fee_per_gas` so that we could extend and call basic validation methods via `super()`, but it sounds confusing :(

Reply:
After dealing with typed transactions, my general take on it was: the *latest* VM should have the cleanest implementation. If a kludge is called for, it should be on the old VMs (like v vs y-parity in the signature, where we started to prefer y-parity as the default/native field). At first glance, this change seems to follow that approach, so I'm 👌🏻 to keep it. I'll comment if I stumble on a better option while reading the rest.